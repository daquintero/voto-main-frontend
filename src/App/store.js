// Reducer Libraries
import { combineReducers, createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension';
import { throttle } from 'lodash';
import { reducer as formReducer } from 'redux-form';

import mapReducer from '../Reusable/Map/redux/reducer';
import detailPageReducer from '../Assembly/DetailedPage/redux/reducer';
// TODO ADD LATER
// import homeReducer from '../Assembly/Social/redux/reducer';
import basicHomeReducer from '../Assembly/Home/redux/reducer';
import welcomeReducer from '../Layout/TopBar/redux/reducer';
import knowledgeBaseReducer from '../Assembly/KnowledgeBase/redux/reducer';
import reusableReducer from '../Reusable/redux/reducer';

// Local Storage
import { loadState, saveState } from './localStorage';


const reducer = combineReducers({
  openPage: detailPageReducer,
  map: mapReducer,
  home: basicHomeReducer,
  form: formReducer,
  welcome: welcomeReducer,
  knowledgeBase: knowledgeBaseReducer,
  reusable: reusableReducer,
});


const storedState = loadState();


const store = createStore(
  reducer,
  storedState,
  composeWithDevTools(applyMiddleware(thunk)),
);

store.subscribe(throttle(() => {
  saveState({
    welcome: store.getState().welcome,
  });
}, 3000));

export default store;
