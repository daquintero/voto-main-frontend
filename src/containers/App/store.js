import { combineReducers, createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension';

// Reusable Reducer Functions
import {
  informativeSnippetReducer,
} from '../Reusable/reusable-redux-index';

const reducer = combineReducers({
  informativeSnippets: informativeSnippetReducer,
});

const store = createStore(
  reducer,
  composeWithDevTools(applyMiddleware(thunk)),
);

export default store;
