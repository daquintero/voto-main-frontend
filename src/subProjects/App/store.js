import { combineReducers, createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension';

// Reusable Reducer Functions
import {
  informativeSnippetReducer,
  individualReducer,
  corruptionCaseReducer,
  organizationReducer,
  financialItemReducer,
  mapReducer,
} from '../Reusable/reusableReduxIndex';

const openPage = combineReducers({
  informativeSnippet: informativeSnippetReducer,
  individual: individualReducer,
  corruptionCase: corruptionCaseReducer,
  organization: organizationReducer,
  financialItem: financialItemReducer,
  map: mapReducer,
});

const reducer = combineReducers({
  openPage,
});

const store = createStore(
  reducer,
  composeWithDevTools(applyMiddleware(thunk)),
);

export default store;
