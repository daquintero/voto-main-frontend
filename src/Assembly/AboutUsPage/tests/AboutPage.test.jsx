// Test for AboutPage Page Component
import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import AboutPage from '../index';
import store from '../../../App/store';

describe('<AboutPage />', () => {
  const initialState = store;
  const match = { params: { id: 1 } } ;

  it(' renders correctly without crashing', () => {
    const tree = renderer
      .create(<>
        <Provider store={initialState}>
          <AboutPage match={match} />
        </Provider>
      </>)
      .toJSON();
    expect(tree)
      .toMatchSnapshot();
  });
});
