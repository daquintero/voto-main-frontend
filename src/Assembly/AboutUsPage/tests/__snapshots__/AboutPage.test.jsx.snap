// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AboutPage />  renders correctly without crashing 1`] = `
Array [
  <div
    className="text-center p-3 py-5 p-md-5 bg-shady-layout container-fluid"
  >
    <div
      className="row"
    >
      <div
        className="col-12"
      >
        <h2>
          #VotoInformado2019
        </h2>
      </div>
    </div>
  </div>,
  <div
    className="py-5 container"
  >
    <div
      className="featured-video"
    >
      <div
        className="video"
      >
        <iframe
          allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
          allowFullScreen={true}
          frameBorder="0"
          src="https://www.youtube.com/embed/gmwbcbNdN-0"
          title="Bienvenido a Voto Informado"
        />
      </div>
    </div>
    <div
      className="my-5 bg-layout no-gutters row"
    >
      <div
        className="p-4 col-md-5"
      >
        <img
          alt=""
          className="w-100"
          src="https://s3.amazonaws.com/votoinformado2019/about/team.jpeg"
        />
      </div>
      <div
        className="p-4 col-md-7"
      >
        <h5
          className="py-2"
        >
          Acerca de Nosotros 
        </h5>
        <p
          style={
            Object {
              "letter-spacing": "0.02em",
            }
          }
        >
          Esta campaña fue creada por el equipo de Praxis, Escuela Callejera y un grupo de ciudadanos, para ciudadanos. El objetivo es hacer docencia para informar y empoderar a los ciudadanos para las elecciones en mayo de 2019.
        </p>
        <p
          style={
            Object {
              "letter-spacing": "0.02em",
            }
          }
        >
          Somos un grupo de jóvenes preocupados por el rumbo en el que va nuestro país. Vale dejar claro que esta es una campaña 
          <strong>
            apartidista
          </strong>
            y fue desarrollada pro-bono por todos los participantes.
        </p>
      </div>
    </div>
    <div
      className="my-5 no-gutters row"
    >
      <div
        className="p-4 col-md-7"
      >
        <h5
          className="py-2"
        >
          Agradecimientos
        </h5>
        <p
          style={
            Object {
              "letter-spacing": "0.02em",
            }
          }
        >
          Queremos darles un agradecimiento especial a todas las personas que participarton en la creación de esta campaña, en la página web, en el Manual de Cívica Electoral, en el Chatbot, las investigaciones y la creación de los perfiles de los candidatos, en los videos, en el diseño y en la visualización de la información, y en todo lo que compone Voto Informado 2019.
        </p>
        <p
          style={
            Object {
              "letter-spacing": "0.02em",
            }
          }
        >
          Apreciamos mucho el apoyo y tiempo de todos, en especial a: los participantes del Investigatón 2018, los participantes de Panamá-Hackea 2018, a los desarrolladores de la plataforma, a Melida Botacio, Angel Corro, Augusto Revello, Franz Delgado, Dario Quintero, James Williams, Leopoldo Alfaro, Emmanuel Lobo, y Cynthia Paolini.
        </p>
      </div>
      <div
        className="p-4 col-md-5"
      >
        <img
          alt="Error"
          className="img-cover"
          src="/img/conteo.png"
        />
      </div>
    </div>
    <div
      className="my-5 p-4 text-center justify-content-center bg-layout no-gutters row"
    >
      <h5
        className="py-4"
      >
        ¿Conocías a Praxis y Escuela Callejera?
      </h5>
      <div
        className="no-gutters row"
      >
        <div
          className="p-2 col-md-4"
        >
          <img
            alt="Error"
            className="w-50 p-2"
            src="https://s3.amazonaws.com/votoinformado2019/about/praxis.jpeg"
          />
        </div>
        <div
          className="p-2 align-middle my-auto col-md-8"
        >
          <p>
            Praxis es un emprendimiento social que busca empoderar a la ciudadanía panameña por medio de la información. Explicamos lo que pasa en Panamá y en el mundo de formas innovadoras de contar la historia.
          </p>
        </div>
      </div>
      <div
        className="no-gutters row"
      >
        <div
          className="p-0 col-md-4"
        >
          <img
            alt="Error"
            className="w-50"
            src="/img/ec.svg"
          />
        </div>
        <div
          className="p-2 my-auto col-md-8"
        >
          <p>
            Escuela callejera es un colectivo que busca crear cultura a través de la educación. Estudiamos temas contemporáneos y buscamos como explicarlos de forma simple.
          </p>
        </div>
      </div>
    </div>
    <div
      className="col"
    >
      <p
        className="lead text-center"
      >
        No olvides seguirnos en Instagram
      </p>
      <p
        className="text-center lead m-1"
      >
        <a
          className="sl"
          href="https://www.instagram.com/votoinformado2019/"
          rel="noreferrer noopener"
          target="_blank"
        >
          <span>
            @votoinformado2019
          </span>
        </a>
      </p>
      <p
        className="text-center m-1"
      >
        <a
          className="lead sl"
          href="https://www.instagram.com/praxispty/"
          rel="noreferrer noopener"
          target="_blank"
        >
          <span>
            @praxispty
          </span>
        </a>
      </p>
      <p
        className="text-center lead m-1"
      >
        <a
          className="sl"
          href="https://www.instagram.com/escuelacallejera/"
          rel="noreferrer noopener"
          target="_blank"
        >
          <span>
            @escuelacallejera
          </span>
        </a>
      </p>
    </div>
    <div
      className="insta-grid"
    >
      <script
        src="https://cdn.lightwidget.com/widgets/lightwidget.js"
      />
      <div
        className="photos-container"
      >
        <iframe
          className="lightwidget-widget"
          scrolling="no"
          src="//lightwidget.com/widgets/554828df384f55d5b8ad277da18957ec.html"
          title="instagram"
        />
      </div>
    </div>
  </div>,
]
`;
